Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1495,7 +1495,13 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrackMasonry(GridTrack
         isNewEntry = true;
 
         // Correct the span as the grid item is coming from another track.
-        span = GridSpan::translatedDefiniteGridSpan(trackIndex, trackIndex + span.integerSpan());
+        unsigned shift = 0;
+        if (span.integerSpan() > 1 && ((span.integerSpan() + trackIndex) > tracks(m_direction).size())) {
+            shift = span.integerSpan() + trackIndex - tracks(m_direction).size();
+            shift = (shift > trackIndex) ? 0 : shift;
```

Comment:
I think I'm having a little trouble understanding this line. What are we trying to do here that causes us to adjust `shift` to 0? (Should we even be comparing `shift` to `trackIndex`?)

Reply:
It does not make sense to shift the grid item below 0. This is designed to protect against that issue. Shifting by 0 does not move the grid item at all.